h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -298,7 +298,7 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
       t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+              H2O.ARGS.nthreads, H2O.getCloudSize(), false /*use new heuristic*/, true);
```

Comment:
This change could go along with Vec change as well. I'd personally do a little test showing this change does not slow us down in edge cases, provided nthreads are user-defined and can easily be a higher number than H2O.NUMCPUS.